### PR TITLE
feat(murojaah): add create and subcreate buttons mockup

### PR DESCRIPTION
**Pull request by RR-Ardhika**
Sample view image:

image

### CODE REVIEW

**Review by taufiqelrahman — APPROVED** [review]

File: src/app/components/CreateButton.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  function toggleShowSubButtons(): void {
+    setIsShowSubButtons((isShowSubButtons: boolean) => !isShowSubButtons);
+  }
+
+  return (
+    <div className="fixed bottom-16 right-4">
+      <div className="flex flex-col gap-4">
+        {isShowSubButtons && showSubCreateButtons()}
+        <button
+          className={createButtonClass}
```
Comment (taufiqelrahman) [inline src/app/components/CreateButton.tsx:47]:
```
className={twMerge(buttonClassnames.button, isShowSubButtons && 'left-[7px]')}
```

File: src/app/components/CreateButton.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      </button>
+    </div>
+  );
+}
+
+export const CreateButton = (): JSX.Element => {
+  const [isShowSubButtons, setIsShowSubButtons]: [
+    boolean,
+    Dispatch<SetStateAction<boolean>>,
+  ] = useState<boolean>(false);
```
Comment (taufiqelrahman) [inline src/app/components/CreateButton.tsx:32]: this typing definition is sufficient
```
const [isShowSubButtons, setIsShowSubButtons] = useState<boolean>(false);
```

Comment (RR-Ardhika) [reply src/app/components/CreateButton.tsx:32]: true, but currently we using the eslint rule that will raise error when we don't specify type of the variables

File: src/app/components/CreateButton.tsx (hunk 1 above)
Comment (RR-Ardhika) [reply src/app/components/CreateButton.tsx:47]: will be updated here https://github.com/RR-Ardhika/murojaah/pull/6/commits/c32d3bc862fe28923a14346f7d29474ac408592b

File: src/app/components/CreateButton.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
+'use client';
+
+import { useState } from 'react';
+
+const buttonClassnames: Record<string, string> = {
+  button:
+    'relative right-1 w-16 h-16 bg-white active:bg-cyan-300 border-2 text-teal border-teal rounded-full',
+  subButton:
+    'p-2 bg-white active:bg-cyan-300 border-2 text-teal border-teal rounded-full',
+};
```
[the remark this answers is not in the document]
Comment (RR-Ardhika) [reply src/app/components/CreateButton.tsx:10]: will be updated here https://github.com/RR-Ardhika/murojaah/pull/6/commits/c32d3bc862fe28923a14346f7d29474ac408592b

**Pull request merged**